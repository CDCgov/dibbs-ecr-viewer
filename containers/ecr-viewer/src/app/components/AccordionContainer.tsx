import React from "react";
import { Accordion, HeadingLevel } from "@trussworks/react-uswds";
import { toKebabCase } from "@/app/utils/format-utils";
import { RenderableNode } from "@/app/utils/data-utils";

export interface AccordionItemProps {
  title: RenderableNode;
  content: React.ReactNode;
  expanded: boolean;
  id: string;
  className?: string;
  headingLevel: HeadingLevel;
  handleToggle?: (event: React.MouseEvent<HTMLButtonElement>) => void;
}

type AccordionContainerProps = {
  accordionItems: AccordionItemProps[];
};

/**
 * Functional component for an accordion container displaying various sections of eCR information.
 * @param props - Props containing FHIR bundle and path mappings.
 * @param props.accordionItems - The list of accordion items.
 * @returns The JSX element representing the accordion container.
 */
const AccordionContainer = ({
  accordionItems,
}: AccordionContainerProps): React.JSX.Element => {
  const items: AccordionItemProps[] = accordionItems.map((item, index) => {
    let formattedTitle = toKebabCase(`${item["title"]}`);
    return {
      ...item,
      id: `${formattedTitle}_${index + 1}`,
      title: <span id={formattedTitle}>{item["title"]}</span>,
    };
  });

  return (
    <Accordion
      className="info-container"
      items={items}
      multiselectable={true}
    />
  );
};
export default AccordionContainer;
