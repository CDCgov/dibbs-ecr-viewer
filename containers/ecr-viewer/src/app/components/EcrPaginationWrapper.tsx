"use client";

import { Label, Select } from "@trussworks/react-uswds";
import React, { ReactNode, useEffect, useState } from "react";
import { Pagination } from "@/app/components/Pagination";
import { usePathname, useRouter, useSearchParams } from "next/navigation";

interface EcrPaginationWrapperProps {
  totalCount: number;
  children: ReactNode;
}

interface UserPreferences {
  itemsPerPage: number;
}

const defaultPreferences = {
  itemsPerPage: 25,
};

/**
 * Renders a list of eCR data with viewer.
 * @param props - The properties passed to the component.
 * @param props.totalCount - Total number of eCRs
 * @param props.children - Contents of the wrapper
 * @returns The JSX element (table) representing the rendered list of eCRs.
 */
const EcrPaginationWrapper = ({
  totalCount,
  children,
}: EcrPaginationWrapperProps) => {
  const searchParams = useSearchParams();
  const router = useRouter();
  const pathname = usePathname();

  const currentPage = Number(searchParams.get("page")) || 1;
  const [userPreferences, setUserPreferences] =
    useState<UserPreferences>(defaultPreferences);

  useEffect(() => {
    const userPreferencesString = localStorage.getItem("userPreferences");
    if (userPreferencesString) {
      const storedPrefs = JSON.parse(userPreferencesString);
      setUserPreferences({ ...defaultPreferences, ...storedPrefs });
    }
  }, []);

  useEffect(() => {
    const current = new URLSearchParams(Array.from(searchParams.entries()));
    current.set("itemsPerPage", userPreferences.itemsPerPage.toString());
    const search = current.toString();
    const query = search ? `?${search}` : "";
    router.push(`${pathname}${query}`);
  }, [userPreferences]);

  const totalPages =
    totalCount > 0 ? Math.ceil(totalCount / userPreferences.itemsPerPage) : 1;

  const startIndex =
    totalCount > 0 ? (currentPage - 1) * userPreferences.itemsPerPage + 1 : 0;

  const endIndex = Math.min(
    currentPage * userPreferences.itemsPerPage,
    totalCount,
  );

  return (
    <div className="main-container height-ecr-library flex-column flex-align-center">
      {children}
      <div className="pagination-bar width-full padding-x-3 padding-y-105 flex-align-self-stretch display-flex flex-align-center">
        <div className="flex-1">
          Showing {startIndex}-{endIndex} of {totalCount} eCRs
        </div>
        <Pagination
          currentPage={currentPage}
          totalPages={totalPages}
          maxSlots={6}
          pathname=""
          className="flex-1"
        />
        <div className="display-flex flex-align-center flex-1 flex-justify-end">
          <Label
            htmlFor="input-select"
            className="margin-top-0 margin-right-1025"
          >
            eCRs per page
          </Label>
          <Select
            id="input-select"
            name="input-select"
            value={userPreferences.itemsPerPage}
            className="styled width-11075 margin-top-0"
            onChange={(e) => {
              const updatedUserPreferences: UserPreferences = {
                ...userPreferences,
                itemsPerPage: +e.target.value,
              };
              setUserPreferences(updatedUserPreferences);
              localStorage.setItem(
                "userPreferences",
                JSON.stringify(updatedUserPreferences),
              );
            }}
          >
            <React.Fragment>
              <option value="25">25</option>
              <option value="50">50</option>
              <option value="75">75</option>
              <option value="100">100</option>
            </React.Fragment>
          </Select>
        </div>
      </div>
    </div>
  );
};

export default EcrPaginationWrapper;
