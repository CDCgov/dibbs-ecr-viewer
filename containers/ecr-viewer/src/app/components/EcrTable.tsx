import React from "react";
import { listEcrData } from "@/app/services/listEcrDataService";
import { EcrTableClient } from "@/app/components/EcrTableClient";
import {
  EcrTableNoData,
  TableHeader,
} from "@/app/components/EcrTableClientBase";
import { DateRangePeriod } from "@/app/utils/date-utils";

/**
 * eCR Table
 * @param props - The properties passed to the component.
 * @param props.currentPage - The current page to be displayed
 * @param props.itemsPerPage - The number of items to be displayed in the table
 * @param props.sortColumn - The column to sort by
 * @param props.sortDirection - The direction to sort by
 * @param props.filterDates - The date range used to filter data
 * @param props.searchTerm - The search term used to list data
 * @param props.filterConditions - (Optional) The reportable condition(s) used to filter the data
 * @param props.headers - Table headers descriptions
 * @returns - eCR Table element
 */
const EcrTable = async ({
  currentPage,
  itemsPerPage,
  sortColumn,
  sortDirection,
  searchTerm,
  filterConditions,
  filterDates,
  headers,
}: {
  currentPage: number;
  itemsPerPage: number;
  sortColumn: string;
  sortDirection: string;
  filterDates: DateRangePeriod;
  searchTerm?: string;
  filterConditions?: string[];
  headers: TableHeader[];
}) => {
  const startIndex = (currentPage - 1) * itemsPerPage;

  const data = await listEcrData(
    startIndex,
    itemsPerPage,
    sortColumn,
    sortDirection,
    filterDates,
    searchTerm,
    filterConditions,
  );

  if (data.length === 0) {
    return <EcrTableNoData headers={headers} />;
  }

  return <EcrTableClient headers={headers} data={data} />;
};

export default EcrTable;
