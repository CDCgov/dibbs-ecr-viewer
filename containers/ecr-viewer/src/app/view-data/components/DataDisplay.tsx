"use client";
import React from "react";
import classNames from "classnames";
import { ToolTipElement } from "@/app/view-data/components/ToolTipElement";
import { FieldValue } from "./FieldValue";

export interface DisplayDataProps {
  title?: string;
  className?: string;
  toolTip?: string;
  value?: React.ReactNode;
  dividerLine?: boolean;
  table?: boolean;
}

/**
 * Functional component for displaying data.
 * @param props - Props for the component.
 * @param props.item - The display data item to be rendered.
 * @param [props.className] - Additional class name for styling purposes.
 * @param [props.themeColor] - Color to use for theming (default="gray").
 * @returns - A React element representing the display of data.
 */
export const DataDisplay: React.FC<{
  item: DisplayDataProps;
  className?: string;
  themeColor?: string;
}> = ({
  item,
  className,
  themeColor = "gray",
}: {
  item: DisplayDataProps;
  className?: string;
  themeColor?: string;
}): React.JSX.Element => {
  item.dividerLine =
    item.dividerLine == null || item.dividerLine == undefined
      ? true
      : item.dividerLine;
  return (
    <div>
      <div className="grid-row">
        <div className="data-title padding-right-1">
          <ToolTipElement toolTip={item.toolTip}>{item.title}</ToolTipElement>
        </div>
        <div
          className={classNames(
            "grid-col maxw7 text-pre-line p-list",
            className,
            item.className ? item.className : "",
          )}
        >
          <FieldValue>{item.value}</FieldValue>
        </div>
      </div>
      {item.dividerLine ? (
        <div className={`section__line_${themeColor}`} />
      ) : (
        ""
      )}
    </div>
  );
};

/**
 * Functional component for displaying (a list of) data tables.
 * @param props - Props containing the item to be displayed.
 * @param props.item - The data item to be displayed.
 * @param props.themeColor - The color used in the line (default "gray").
 * @returns The JSX element representing the data table display.
 */
export const DataTableDisplay: React.FC<{
  item: DisplayDataProps;
  themeColor?: string;
}> = ({ item, themeColor = "gray" }): React.JSX.Element => {
  const dividerLine = item.dividerLine ?? true;
  return (
    <>
      <div className="grid-row">
        <div className="grid-col-auto width-full text-pre-line">
          {item.value}
        </div>
      </div>
      {dividerLine && <div className={`section__line_${themeColor}`} />}
    </>
  );
};
