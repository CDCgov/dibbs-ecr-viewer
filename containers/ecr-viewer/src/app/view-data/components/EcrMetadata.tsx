import React from "react";

import { Table } from "@trussworks/react-uswds";

import {
  ERSDWarning,
  ReportableConditions,
} from "@/app/services/ecrMetadataService";
import {
  AccordionSection,
  AccordionSubSection,
} from "@/app/view-data/component-utils";

import { DataDisplay, DisplayDataProps } from "./DataDisplay";
import { ToolTipElement } from "./ToolTipElement";

interface EcrMetadataProps {
  rrDetails: ReportableConditions;
  eicrDetails: DisplayDataProps[];
  eRSDWarnings: ERSDWarning[];
  eCRCustodianDetails: DisplayDataProps[];
  eicrAuthorDetails: DisplayDataProps[][];
}

/**
 * Functional component for displaying eCR metadata.
 * @param props - Props containing eCR metadata.
 * @param props.rrDetails - The reportable conditions details.
 * @param props.eicrDetails - The eICR details.
 * @param props.eRSDWarnings - The eRSD warnings.
 * @param props.eCRCustodianDetails - The eCR custodian details.
 * @param props.eicrAuthorDetails - The eICR author details.
 * @returns The JSX element representing the eCR metadata.
 */
const EcrMetadata = ({
  rrDetails,
  eicrDetails,
  eRSDWarnings,
  eCRCustodianDetails,
  eicrAuthorDetails,
}: EcrMetadataProps) => {
  return (
    <AccordionSection>
      <AccordionSubSection title="RR Details">
        <ReportabilitySummary rrDetails={rrDetails} />
        {eRSDWarnings?.length > 0 && (
          <div>
            <div className="section__line_gray"></div>
            <Table
              bordered={false}
              className="ersd-table fixed-table border-top border-left border-right border-bottom"
              fixed={true}
              fullWidth={true}
            >
              <caption>
                <ToolTipElement toolTip="Can be used to help you identify healthcare providers that need to update their eRSD (Electronic Reporting and Surveillance Distribution) version.">
                  eRSD Warnings
                </ToolTipElement>
              </caption>
              <thead>
                <tr>
                  <th>Warning</th>
                  <th>Version in Use</th>
                  <th>Expected Version</th>
                  <th>Suggested Solution</th>
                </tr>
              </thead>
              <tbody>
                {Array.isArray(eRSDWarnings) &&
                  eRSDWarnings.map((warningItem, index) => (
                    <tr key={index}>
                      <td className="padding-105">{warningItem.warning}</td>
                      <td className="padding-105">{warningItem.versionUsed}</td>
                      <td className="padding-105">
                        {warningItem.expectedVersion}
                      </td>
                      <td className="padding-105">
                        {warningItem.suggestedSolution}
                      </td>
                    </tr>
                  ))}
              </tbody>
            </Table>
            <div className="section__line_gray"></div>
          </div>
        )}
      </AccordionSubSection>

      <AccordionSubSection title="eICR Details">
        {eicrDetails.map((item, index) => {
          return <DataDisplay item={item} key={index} />;
        })}
      </AccordionSubSection>

      {eicrAuthorDetails?.map((authorDetailsDisplayProps, i) => {
        return (
          <AccordionSubSection
            key={`author-${i}`}
            title="eICR Author Details for Practitioner"
          >
            {authorDetailsDisplayProps.map((item, j) => {
              return <DataDisplay item={item} key={`author-details-${j}`} />;
            })}
          </AccordionSubSection>
        );
      })}

      <AccordionSubSection
        title="eICR Custodian Details"
        toolTip="Person or organization that generated the eICR Document."
      >
        {eCRCustodianDetails.map((item, index) => {
          return <DataDisplay item={item} key={index} />;
        })}
      </AccordionSubSection>
    </AccordionSection>
  );
};

type ReportabilitySummaryProps = Pick<EcrMetadataProps, "rrDetails">;

const ReportabilitySummary: React.FC<ReportabilitySummaryProps> = ({
  rrDetails,
}) => {
  const rows = useConvertDictionaryToRows(rrDetails);

  if (rows.length === 0) {
    return null;
  }

  return (
    <Table
      bordered={true}
      caption="Reportability Summary"
      className="rrTable"
      fixed={true}
      fullWidth={true}
    >
      <thead>
        <tr>
          <th className="width-25p">
            <ToolTipElement toolTip="List of conditions that caused this eCR to be sent to your jurisdiction based on the rules set up for routing eCRs by your jurisdiction in RCKMS (Reportable Condition Knowledge Management System). Can include multiple Reportable Conditions for one eCR.">
              Reportable Condition
            </ToolTipElement>
          </th>
          <th>
            <ToolTipElement toolTip="Reason(s) that this eCR was sent for this condition. Corresponds to your jurisdiction's rules for routing eCRs in RCKMS (Reportable Condition Knowledge Management System).">
              RCKMS Rule Summary
            </ToolTipElement>
          </th>
          <th className="width-25p">
            <ToolTipElement toolTip="List of jurisdictions this eCR was sent to. Can include multiple jurisdictions depending on provider location, patient address, and jurisdictions onboarded to eCR.">
              Jurisdiction Sent eCR
            </ToolTipElement>
          </th>
        </tr>
      </thead>
      <tbody>
        {rows.map(({ key, condition, trigger, location }) => (
          <tr key={key}>
            {condition ? (
              <td rowSpan={condition.rowSpan}>{condition.value}</td>
            ) : null}
            {trigger ? (
              <td rowSpan={trigger.rowSpan}>{trigger.value}</td>
            ) : null}
            <td>{location}</td>
          </tr>
        ))}
      </tbody>
    </Table>
  );
};

interface ReportableConditionData {
  value: string;
  rowSpan: number;
}

interface ReportableConditionRow {
  key: string;
  condition: ReportableConditionData | null;
  trigger: ReportableConditionData | null;
  location: string;
}

const useConvertDictionaryToRows = (dictionary: ReportableConditions) => {
  if (!dictionary) return [];
  const rows: ReportableConditionRow[] = [];

  Object.entries(dictionary).forEach(([condition, triggers], _) => {
    Object.entries(triggers).forEach(([trigger, locations], triggerIndex) => {
      const locationsArray = Array.from(locations);
      locationsArray.forEach((location, locationIndex) => {
        const isConditionRow = triggerIndex === 0 && locationIndex === 0;
        const isTriggerRow = locationIndex === 0;

        const conditionRowSpan = Object.keys(triggers).reduce(
          (acc, key) => acc + Array.from(triggers[key]).length,
          0,
        );
        const triggerRowSpan = locationsArray.length;

        const row: ReportableConditionRow = {
          key: `${condition}-${trigger}-${location}`,
          condition: isConditionRow
            ? {
                value: condition,
                rowSpan: conditionRowSpan,
              }
            : null,
          trigger: isTriggerRow
            ? {
                value: trigger,
                rowSpan: triggerRowSpan,
              }
            : null,
          location: location ?? "",
        };

        rows.push(row);
      });
    });
  });

  return rows;
};

export default EcrMetadata;
