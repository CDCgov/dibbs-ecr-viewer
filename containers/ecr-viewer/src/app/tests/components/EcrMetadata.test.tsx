import React from "react";

import { render } from "@testing-library/react";
import { axe } from "jest-axe";

import {
  ERSDWarning,
  ReportableConditions,
} from "@/app/services/ecrMetadataService";
import { DisplayDataProps } from "@/app/view-data/components/DataDisplay";
import EcrMetadata from "@/app/view-data/components/EcrMetadata";

describe("ECR Metadata", () => {
  let container: HTMLElement;
  beforeAll(() => {
    const mockChildMethod = jest.fn();
    jest.spyOn(React, "useRef").mockReturnValue({
      current: {
        childMethod: mockChildMethod,
      },
    });
    const rrConditionsList: ReportableConditions = {
      "Disease caused by severe acute respiratory syndrome coronavirus 2(disorder)":
        {
          "Detection of SARS-CoV-2 nucleic acid in a clinical or post-mortem specimen by any method":
            new Set([
              "Coruscant Department of Public Health",
              "Coruscant City Department of Public Health",
            ]),
          "Close contact in the 14 days prior to onset of symptoms with a confirmed or probable case of COVID-19 (Partially implemented as exposure with no timeframe parameters)":
            new Set(["Coruscant City Department of Public Health"]),
          "COVID-19 (as a diagnosis or active problem)": new Set([
            "Coruscant City Department of Public Health",
          ]),
        },
    };
    const eicrDetails: DisplayDataProps[] = [
      {
        title: "eICR Identifier",
        value: "1dd10047-2207-4eac-a993-0f706c88be5d",
      },
      {
        title: "Date/Time eCR Created",
        value: "2022-05-14T12:56:38Z",
      },
      { title: "eICR Release Version", value: "R1.1 (2016-12-01)" },
      {
        title: "EHR Software Name",
        value: "Epic - Version 10.1",
      },
      { title: "EHR Manufacturer Model Name", value: "Epic - Version 10.1" },
    ];

    const eRSDWarnings: ERSDWarning[] = [
      {
        warning:
          "Sending organization is using an malformed eRSD (RCTC) version",
        versionUsed: "2020-06-23",
        expectedVersion:
          "Sending organization should be using one of the following: 2023-10-06, 1.2.2.0, 3.x.x.x.",
        suggestedSolution:
          "The trigger code version your organization is using could not be determined. The trigger codes may be out date. Please have your EHR administrators update the version format for complete eCR functioning.",
      },
    ];

    const ecrCustodianDetails: DisplayDataProps[] = [
      {
        title: "Custodian ID",
        value: "112233445566",
      },
      {
        title: "Custodian Name",
        value: "Hoth University Medical Center",
      },
      {
        title: "Custodian Address",
        value: "5555 Snowy Ave\nSnowtown, TN\n00123, USA",
      },
      {
        title: "Custodian Contact",
        value: "Work 1-555-555-5555",
      },
    ];

    const eicrAuthorDetails = [
      [
        {
          title: "Author Name",
          value: "Dr. R2D2",
        },
        {
          title: "Author Address",
          value: "1 Droid st",
        },
        {
          title: "Author Contact",
          value: "(555)555-5555",
        },
        {
          title: "Author Facility Name",
          value: "Echo Base Medical Facility",
        },
        {
          title: "Author Facility Address",
          value: "5555 Echo Base Drive\nEcho Base, CA\n00123, USA",
        },
        {
          title: "Author Facility Contact",
          value: "555-555-5555",
        },
      ],
    ];

    container = render(
      <EcrMetadata
        eicrDetails={eicrDetails}
        rrDetails={rrConditionsList}
        eRSDWarnings={eRSDWarnings}
        eCRCustodianDetails={ecrCustodianDetails}
        eicrAuthorDetails={eicrAuthorDetails}
      />,
    ).container;
  });
  it("should match snapshot", () => {
    expect(container).toMatchSnapshot();
  });
  it("should pass accessibility test", async () => {
    expect(await axe(container)).toHaveNoViolations();
  });
});
