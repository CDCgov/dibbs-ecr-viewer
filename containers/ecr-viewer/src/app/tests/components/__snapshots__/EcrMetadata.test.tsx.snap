// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ECR Metadata should match snapshot 1`] = `
<div>
  <div
    class="margin-top-0"
  >
    <div
      class="padding-bottom-3"
    >
      <div
        class="usa-summary-box__body"
      >
        <div
          class="gutter-3"
        >
          <h4
            class="usa-summary-box__heading padding-y-105"
            id="rr-details"
          >
            RR Details
          </h4>
          <div
            class="usa-summary-box__text"
          >
            <table
              class="usa-table width-full _fixed_1khz6_1 rrTable"
              data-testid="table"
            >
              <caption>
                Reportability Summary
              </caption>
              <thead>
                <tr>
                  <th
                    class="width-25p"
                  >
                    <span
                      class="usa-tooltip"
                      data-testid="tooltipWrapper"
                    >
                      <div
                        aria-describedby="[object Object]"
                        class="usa-tooltip__trigger usa-tooltip"
                        data-testid="triggerElement"
                        tabindex="0"
                        title=""
                      >
                        Reportable Condition
                      </div>
                      <span
                        aria-hidden="true"
                        class="usa-tooltip__body"
                        data-testid="tooltipBody"
                        id="[object Object]"
                        role="tooltip"
                        title="List of conditions that caused this eCR to be sent to your jurisdiction based on the rules set up for routing eCRs by your jurisdiction in RCKMS (Reportable Condition Knowledge Management System). Can include multiple Reportable Conditions for one eCR."
                      >
                        List of conditions that caused this eCR to be sent to your jurisdiction based on the rules set up for routing eCRs by your jurisdiction in RCKMS (Reportable Condition Knowledge Management System). Can include multiple Reportable Conditions for one eCR.
                      </span>
                    </span>
                  </th>
                  <th>
                    <span
                      class="usa-tooltip"
                      data-testid="tooltipWrapper"
                    >
                      <div
                        aria-describedby="[object Object]"
                        class="usa-tooltip__trigger usa-tooltip"
                        data-testid="triggerElement"
                        tabindex="0"
                        title=""
                      >
                        RCKMS Rule Summary
                      </div>
                      <span
                        aria-hidden="true"
                        class="usa-tooltip__body"
                        data-testid="tooltipBody"
                        id="[object Object]"
                        role="tooltip"
                        title="Reason(s) that this eCR was sent for this condition. Corresponds to your jurisdiction's rules for routing eCRs in RCKMS (Reportable Condition Knowledge Management System)."
                      >
                        Reason(s) that this eCR was sent for this condition. Corresponds to your jurisdiction's rules for routing eCRs in RCKMS (Reportable Condition Knowledge Management System).
                      </span>
                    </span>
                  </th>
                  <th
                    class="width-25p"
                  >
                    <span
                      class="usa-tooltip"
                      data-testid="tooltipWrapper"
                    >
                      <div
                        aria-describedby="[object Object]"
                        class="usa-tooltip__trigger usa-tooltip"
                        data-testid="triggerElement"
                        tabindex="0"
                        title=""
                      >
                        Jurisdiction Sent eCR
                      </div>
                      <span
                        aria-hidden="true"
                        class="usa-tooltip__body"
                        data-testid="tooltipBody"
                        id="[object Object]"
                        role="tooltip"
                        title="List of jurisdictions this eCR was sent to. Can include multiple jurisdictions depending on provider location, patient address, and jurisdictions onboarded to eCR."
                      >
                        List of jurisdictions this eCR was sent to. Can include multiple jurisdictions depending on provider location, patient address, and jurisdictions onboarded to eCR.
                      </span>
                    </span>
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td
                    rowspan="4"
                  >
                    Disease caused by severe acute respiratory syndrome coronavirus 2(disorder)
                  </td>
                  <td
                    rowspan="2"
                  >
                    Detection of SARS-CoV-2 nucleic acid in a clinical or post-mortem specimen by any method
                  </td>
                  <td>
                    Coruscant Department of Public Health
                  </td>
                </tr>
                <tr>
                  <td>
                    Coruscant City Department of Public Health
                  </td>
                </tr>
                <tr>
                  <td
                    rowspan="1"
                  >
                    Close contact in the 14 days prior to onset of symptoms with a confirmed or probable case of COVID-19 (Partially implemented as exposure with no timeframe parameters)
                  </td>
                  <td>
                    Coruscant City Department of Public Health
                  </td>
                </tr>
                <tr>
                  <td
                    rowspan="1"
                  >
                    COVID-19 (as a diagnosis or active problem)
                  </td>
                  <td>
                    Coruscant City Department of Public Health
                  </td>
                </tr>
              </tbody>
            </table>
            <div>
              <div
                class="section__line_gray"
              />
              <table
                class="usa-table usa-table--borderless width-full _fixed_1khz6_1 ersd-table fixed-table border-top border-left border-right border-bottom"
                data-testid="table"
              >
                <caption>
                  <span
                    class="usa-tooltip"
                    data-testid="tooltipWrapper"
                  >
                    <div
                      aria-describedby="[object Object]"
                      class="usa-tooltip__trigger usa-tooltip"
                      data-testid="triggerElement"
                      tabindex="0"
                      title=""
                    >
                      eRSD Warnings
                    </div>
                    <span
                      aria-hidden="true"
                      class="usa-tooltip__body"
                      data-testid="tooltipBody"
                      id="[object Object]"
                      role="tooltip"
                      title="Can be used to help you identify healthcare providers that need to update their eRSD (Electronic Reporting and Surveillance Distribution) version."
                    >
                      Can be used to help you identify healthcare providers that need to update their eRSD (Electronic Reporting and Surveillance Distribution) version.
                    </span>
                  </span>
                </caption>
                <thead>
                  <tr>
                    <th>
                      Warning
                    </th>
                    <th>
                      Version in Use
                    </th>
                    <th>
                      Expected Version
                    </th>
                    <th>
                      Suggested Solution
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td
                      class="padding-105"
                    >
                      Sending organization is using an malformed eRSD (RCTC) version
                    </td>
                    <td
                      class="padding-105"
                    >
                      2020-06-23
                    </td>
                    <td
                      class="padding-105"
                    >
                      Sending organization should be using one of the following: 2023-10-06, 1.2.2.0, 3.x.x.x.
                    </td>
                    <td
                      class="padding-105"
                    >
                      The trigger code version your organization is using could not be determined. The trigger codes may be out date. Please have your EHR administrators update the version format for complete eCR functioning.
                    </td>
                  </tr>
                </tbody>
              </table>
              <div
                class="section__line_gray"
              />
            </div>
          </div>
        </div>
        <div
          class="gutter-3"
        >
          <h4
            class="usa-summary-box__heading padding-y-105"
            id="eicr-details"
          >
            eICR Details
          </h4>
          <div
            class="usa-summary-box__text"
          >
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  eICR Identifier
                </div>
                <div
                  class="grid-col maxw7 text-pre-line p-list"
                >
                  1dd10047-2207-4eac-a993-0f706c88be5d
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Date/Time eCR Created
                </div>
                <div
                  class="grid-col maxw7 text-pre-line p-list"
                >
                  2022-05-14T12:56:38Z
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  eICR Release Version
                </div>
                <div
                  class="grid-col maxw7 text-pre-line p-list"
                >
                  R1.1 (2016-12-01)
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  EHR Software Name
                </div>
                <div
                  class="grid-col maxw7 text-pre-line p-list"
                >
                  Epic - Version 10.1
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  EHR Manufacturer Model Name
                </div>
                <div
                  class="grid-col maxw7 text-pre-line p-list"
                >
                  Epic - Version 10.1
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
          </div>
        </div>
        <div
          class="gutter-3"
        >
          <h4
            class="usa-summary-box__heading padding-y-105"
            id="eicr-author-details-for-practitioner"
          >
            eICR Author Details for Practitioner
          </h4>
          <div
            class="usa-summary-box__text"
          >
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Author Name
                </div>
                <div
                  class="grid-col maxw7 text-pre-line p-list"
                >
                  Dr. R2D2
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Author Address
                </div>
                <div
                  class="grid-col maxw7 text-pre-line p-list"
                >
                  1 Droid st
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Author Contact
                </div>
                <div
                  class="grid-col maxw7 text-pre-line p-list"
                >
                  (555)555-5555
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Author Facility Name
                </div>
                <div
                  class="grid-col maxw7 text-pre-line p-list"
                >
                  Echo Base Medical Facility
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Author Facility Address
                </div>
                <div
                  class="grid-col maxw7 text-pre-line p-list"
                >
                  5555 Echo Base Drive
Echo Base, CA
00123, USA
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Author Facility Contact
                </div>
                <div
                  class="grid-col maxw7 text-pre-line p-list"
                >
                  555-555-5555
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
          </div>
        </div>
        <div
          class="gutter-3"
        >
          <span
            class="usa-tooltip"
            data-testid="tooltipWrapper"
          >
            <div
              aria-describedby="[object Object]"
              class="usa-tooltip__trigger usa-tooltip short-tooltip"
              data-testid="triggerElement"
              tabindex="0"
              title=""
            >
              <h4
                class="usa-summary-box__heading padding-y-105"
                id="eicr-custodian-details"
              >
                eICR Custodian Details
              </h4>
            </div>
            <span
              aria-hidden="true"
              class="usa-tooltip__body"
              data-testid="tooltipBody"
              id="[object Object]"
              role="tooltip"
              title="Person or organization that generated the eICR Document."
            >
              Person or organization that generated the eICR Document.
            </span>
          </span>
          <div
            class="usa-summary-box__text"
          >
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Custodian ID
                </div>
                <div
                  class="grid-col maxw7 text-pre-line p-list"
                >
                  112233445566
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Custodian Name
                </div>
                <div
                  class="grid-col maxw7 text-pre-line p-list"
                >
                  Hoth University Medical Center
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Custodian Address
                </div>
                <div
                  class="grid-col maxw7 text-pre-line p-list"
                >
                  5555 Snowy Ave
Snowtown, TN
00123, USA
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Custodian Contact
                </div>
                <div
                  class="grid-col maxw7 text-pre-line p-list"
                >
                  Work 1-555-555-5555
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
